source/isaaclab_mimic/isaaclab_mimic/motion_planners/base_motion_planner.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2024-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from abc import ABC, abstractmethod
+from typing import Any
+
+from isaaclab.assets import Articulation
+from isaaclab.envs.manager_based_env import ManagerBasedEnv
+
+
+class MotionPlanner(ABC):
```

Comment:
If this is an abstract class, it is good to have it clearly defined in the name.

Additionally, the file name should be `motion_planner_base.py` to stay consistent with the rest of the naming.

```suggestion
class MotionPlannerBase(ABC):
```